Core/GDCore/IDE/Events/EventsVariablesFinder.cpp

```diff
@@ -203,6 +207,32 @@ std::set<gd::String> EventsVariablesFinder::FindArgumentsInInstructions(
   return results;
 }
 
+std::set<gd::String> EventsVariablesFinder::FindArgumentsInEventsAndDependencies(
+    const gd::Platform& platform,
+    const gd::Project& project,
+    const gd::Layout& layout,
+    const gd::EventsList& events,
+    const gd::String& parameterType,
+    const gd::String& objectName) {
+  std::set<gd::String> results;
+
+  std::set<gd::String> results2 = FindArgumentsInEvents(
+      platform, project, layout, events, parameterType, objectName);
+  results.insert(results2.begin(), results2.end());
+
+  DependenciesAnalyzer dependenciesAnalyzer = DependenciesAnalyzer(project, layout);
+  dependenciesAnalyzer.Analyze();
+  for (gd::String externalEventName : dependenciesAnalyzer.GetExternalEventsDependencies()) {

```

Comment:
Use a a const ref (const gd::String& externalEventName) to avoid a copy of each string at each iteration